source/Calamari/Kubernetes/ResourceStatus/ResourceUpdateReporter.cs

```diff
@@ -64,11 +65,21 @@ private static IEnumerable<Resource> GetRemovedResources(IDictionary<string, Res
         
         private void SendServiceMessage(Resource resource, bool removed, int checkCount)
         {
+            var actionNumber = variables.Get("Octopus.Action.Number", string.Empty);
+            var stepNumber = variables.Get("Octopus.Step.Number");
+            var stepName = variables.Get("Octopus.Step.Name");
+            
+            if (actionNumber.IndexOf(".", StringComparison.Ordinal) > 0)
+            {
+                stepNumber = actionNumber;
+                stepName = variables.Get("Octopus.Action.Name");
+            }
```

Comment:
Looks good, but it seems if there is only one action under a step, then the action number and action name will just be the step number and step name, so perhaps we can directly use those. See in the screenshot that the action name and step name, action number and step number are the same

<img width="753" alt="Screenshot 2023-07-03 at 5 21 13 PM" src="https://github.com/OctopusDeploy/Calamari/assets/97418140/e371a513-6930-4483-b012-fb4ea2005d90">

Reply:
So we are only changing the naming behaviour for child steps, so I am not sure I understand what you saying @YihaoOct ?

Reply:
I mean, why not directly use `Octopus.Action.Number` and `Octopus.Action.Name` instead of those for steps for all?

Reply:
Ah, because `Octopus.Action.Name` may not be the same as `Octopus.Step.Name`.
Via the API, they can be different 😉

Reply:
We want those names to be the same as when the users are looking at the process itself.